csvdiff/__init__.py

```diff
@@ -31,17 +31,26 @@
 EXIT_ERROR = 2
 
 
+def diff(from_csv, to_csv, index_columns, sep=',', ignored_columns=None):
+    """
+    Diff two CSVs, returning the patch which transforms one into the
+    other.
+    """
+    return diff_files(io.StringIO(from_csv), io.StringIO(to_csv),
+                      index_columns, sep, ignored_columns)
+
+
```

Comment:
You want to diff serialised CSV strings? Seems like too much a corner case to be in the API.

We have already: `diff_files` and `diff_records`. We could add a `diff_pandas` though, if that's where you're coming from.

Then you would do something like:

```python
def diff_pandas(from_df, to_df, index_columns, ...):
    from_records = from_df.to_dict(orient='records')
    ...
    return diff_records(from_records, to_records, ...)
```

I reckon that would be really useful. Would that work for your use case?